### PR TITLE
Add new version of MGTransferGlobalCoarseningTools::create_geometric_coarsening_sequence() that takes partitioner

### PR DESCRIPTION
**Pull request by peterrum**
~... depends on #12312 and #12348.~

If #12291 would still be open, this would close it :)

### CODE REVIEW

**Comment by peterrum** [comment]
I have rebased this branch.

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h [hunk 1]
```diff
@@ -2535,10 +2616,6 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::
 
   const bool use_src_inplace = this->vec_fine.size() == 0;
   const auto vec_fine_ptr    = use_src_inplace ? &src : &this->vec_fine;
-
-  if (use_src_inplace == false)
-    this->vec_fine.copy_locally_owned_data_from(src);
-
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h:2541]: Is this code deleted by accident?

File: tests/fullydistributed_grids/repartitioning_03.cc [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+template <int dim, int spacedim>
+void
+output_grid(
+  const std::vector<std::shared_ptr<const Triangulation<dim, spacedim>>> &trias,
+  const std::string &                                                     label)
+{
+  const auto comm    = trias.front()->get_communicator();
+  const auto my_rank = Utilities::MPI::this_mpi_process(comm);
+
```
Comment (kronbichler) [inline tests/fullydistributed_grids/repartitioning_03.cc:77]: Would you mind printing the label to the deallog here to get some more structure into the test output?

**Comment by peterrum** [comment]
@kronbichler Fixed!

**Review by kronbichler — APPROVED** [review]

**Comment by kronbichler** [comment]
Can you please rebase, then I think we can merge this PR.

**Comment by peterrum** [comment]
Done!

**Review by kronbichler — APPROVED** [review]

**Pull request merged**